### PR TITLE
Stories and Actors

### PR DESCRIPTION
**Pull request by Iamgoofball**
### Please read the design doc at https://hackmd.io/@Dsm-BOZJQquXV0Z7ShxJQg/S1C078hCo

I put this PR up as @jlsnow301 expressed interest in the proposal.

## Changelog

🆑 Zonespace and Iamgoofball, passport sprites by Flavrius, stories written by Oscar Gilmour
add: Adds Stories to the game, a new system designed to enhance the roleplaying experience and breathe life, drama, and soap opera plots into the station!
/🆑

# Stories to implement:
All of these stories are finished and written by Oscar Gilmour, a professional writer. Just working on getting them punched into the code. I will probably commission more after launch.

## Season 1
- [X] Ominous Past 
- [X] Zoldorf's Apprentice
- [X] Auteurs In Space
- [x] Sample Smuggler
- [x] The Quest for The Perfect Brew
- [x] Alternative Medicine
- [x] Regulation Station
- [x] Visions
- [x] Guardian Angel
- [x] Guardnapped
- [x] Worldjumper
- [ ] The Show Must Go On
- [ ] No More Strings
- [ ] Peace Talks
- [ ] Galactica News
- [ ] Memory Purged
- [ ] Goodbye, Space Station 13
- [ ] The Schemes of The Necromancer
- [ ] Lux Leather versus The Justices

## Season 2
- [ ] Head Assistant
- [ ] Triangular Profits
- [ ] Special Delivery
- [ ] Codewords
- [ ] Tell me lies, tell me Syndicate lies
- [ ] Senate Business
- [ ] Peaceful Protest
- [ ] Technophobe
- [ ] Gambler's Gambling Gambit
- [ ] Loose Cannon
- [ ] Organ Dealer
- [ ] BOMBS!
- [ ] Poker Night
- [ ] People Against Oxygen
- [ ] Golden Chef
- [ ] Boardline Impossible
- [ ] Salvation
- [ ] Error 99
- [ ] Of Many Faces
- [ ] Good Workers Follow Orders

### CODE REVIEW

**Comment by Kapu1178** [comment]
Nothing says roleplay like the game telling you how to roleplay

File: code/modules/client/preferences/story_participant.dm
Comment (Mothblocks) [inline code/modules/client/preferences/story_participant.dm:1]: `story_pref` -> `story_participant`

File: code/modules/client/preferences/story_participant.dm
Comment (Mothblocks) [inline code/modules/client/preferences/story_participant.dm:4]: Same here

File: code/modules/stories_system/code/actor/_actor.dm
Comment (Mothblocks) [inline code/modules/stories_system/code/actor/_actor.dm:1]: Need to document this

File: code/modules/stories_system/code/actor/_actor.dm
Comment (Mothblocks) [inline code/modules/stories_system/code/actor/_actor.dm:26]: Don't need this

File: code/modules/stories_system/code/actor/crew_actors.dm
Comment (Mothblocks) [inline code/modules/stories_system/code/actor/crew_actors.dm:1]: All these base classes need documentation

File: code/modules/stories_system/code/actor/ghost/arrivals_spawn.dm
Comment (Mothblocks) [inline code/modules/stories_system/code/actor/ghost/arrivals_spawn.dm:211]: Instead of using "is", you should be storing their `p_are()` result.

File: code/modules/stories_system/code/actor/ghost/ghost_actors.dm
Comment (Mothblocks) [inline code/modules/stories_system/code/actor/ghost/ghost_actors.dm:39]:
```suggestion
		ASSERT(landmark, "failed to find an actor spawn ID landmark after making a template that should've assigned one!")
```

File: code/modules/stories_system/code/actor/ghost/ghost_actors.dm
Comment (Mothblocks) [inline code/modules/stories_system/code/actor/ghost/ghost_actors.dm:26]:
```suggestion
		ASSERT(reservation_area, "failed to reserve an area for actor spawning")
```

File: code/modules/stories_system/code/actor/ghost/ghost_actors.dm
Comment (Mothblocks) [inline code/modules/stories_system/code/actor/ghost/ghost_actors.dm:49]:
```suggestion
	temporary_human.balloon_alert(temporary_human, "[DisplayTimeText(delay_time)] until you're sent in")
```

Comment (Mothblocks) [inline code/modules/stories_system/code/actor/ghost/ghost_actors.dm:49]: I think unless this has a UI element (it should have a UI element), this should be a to_chat since it's pretty important. But I think you should make a UI element instead.

File: code/modules/stories_system/code/actor/ghost/ghost_actors.dm
Comment (Mothblocks) [inline code/modules/stories_system/code/actor/ghost/ghost_actors.dm:69]: Don't loop over areas. Use `get_area_contents()` or whatever it's called instead.

File: code/modules/stories_system/code/actor/ghost/maint_spawn.dm
Comment (Mothblocks) [inline code/modules/stories_system/code/actor/ghost/maint_spawn.dm:45]: For all of these, indent up one.

File: code/modules/stories_system/code/actor/ghost/teleporting_spawn.dm
Comment (Mothblocks) [inline code/modules/stories_system/code/actor/ghost/teleporting_spawn.dm:18]: Named parameters for all numbers

File: code/modules/stories_system/code/actor/ghost/teleporting_spawn.dm
Comment (Mothblocks) [inline code/modules/stories_system/code/actor/ghost/teleporting_spawn.dm:20]: Named parameter for TRUE

File: code/modules/stories_system/code/config.dm
Comment (Mothblocks) [inline code/modules/stories_system/code/config.dm:1]: Draft until all these TODOs are done

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Didn't look at all of it and probalby won't

**Comment by jlsnow301** [comment]
Nice. Gimmick roles were one of my favorite additions to Bee, would love to see how they'd fare here

**Comment by DAKKA-WAAAGH** [comment]
This is a very gentle way of introducing players to roleplay. Love it.

Will it be very easy for players to add onto it should it make it in?

**Comment by wsock** [comment]
You're honestly a weirdo with this stuff.

File: code/modules/stories_system/code/actor/ghost/ghost_actors.dm
Comment (Iamgoofball) [reply code/modules/stories_system/code/actor/ghost/ghost_actors.dm:49]: agreed, will do this

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/game_preferences/story_participant.tsx
Comment (Mothblocks) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/game_preferences/story_participant.tsx:4]: This feels like a pretty huge thing to lock behind a small preference. I think it should be visible in the antagonist menu with everything else even if it does not contribute to Dynamic.

File: code/modules/client/preferences/story_participant.dm
Comment (Mothblocks) [inline code/modules/client/preferences/story_participant.dm:3]: Off by default. Most people are going to ignore this because they don't like being directed how to roleplay.

File: code/modules/stories_system/code/subsystem.dm
Comment (Mothblocks) [inline code/modules/stories_system/code/subsystem.dm:22]: Use the `abstract` pattern instead

**Comment by BeeSting12** [comment]
Didn't catch this in the design doc but had a couple questions. Are roundstart characters spawning as actors in these stories, are they midround antags, or combination? I'm also assuming these roles will have their own rules somewhere between a crewmembers and an antagonist, similar to lavaland where it's basically follow your flavor text. I don't look forward to seeing 42 policy threads about each story. How many stories will spawn per round? Is it meant to be something rare, or will everyone with the preference get one?

File: code/modules/client/preferences/story_participant.dm
Comment (Iamgoofball) [reply code/modules/client/preferences/story_participant.dm:3]: i'm not turning it off by default, that will massively reduce uptake of the feature

**Comment by Iamgoofball** [comment]
> Are roundstart characters spawning as actors in these stories

We execute a story roundstart, so yes, you can end up as a crew actor(ie. normal crewmember but an actor) or pulled to be one of the ghost-spawn actors.

>are they midround antags

Stories can execute midround, but they are unrelated to the antagonist system.

> I'm also assuming these roles will have their own rules somewhere between a crewmembers and an antagonist, similar to lavaland where it's basically follow your flavor text.

That's up to the admins, we don't do policy here at the codebase, separation of server and codebase is absolute etc.

>How many stories will spawn per round?

There's a budget system and stories are sorted by level of impactfulness, which determines how much of the budget they take up. Specific numbers are a work in progress as we fine tune things.

**Comment by Loafin34** [comment]
I love this, I can only see one problem: If there's a set amount of scenes, wouldn't people get tired of it? I'm just hoping the stories are vague enough so when a person rolls an actor they aren't groaning since "Oh, I got the organ dealer one, FUCK, I wanted the guard one." Good luck, goofball.

**Comment by SomeoneYouProbablyKnow** [comment]
What's the order of operation on picking actors? Can an actor become an antagonist as well?

Is there a system in place to hook the crew into those stories? Fugitives (the basic ones, i feel like they are comparable to this system) don't impact the round because people don't even know they spawned, that could be an issue.

With how sec is right now they might not care about non-antagonists the same way they don't care about drug dealing? Or do you consider it just an admin issue?

**Comment by celulamp** [comment]
This is actually pretty cool but it would only work on manual

**Comment by Iamgoofball** [comment]
@celulamp all our servers are RP servers

**Comment by ChungusGamer666** [comment]
> @celulamp all our servers are RP servers

The families pipeline all over again

**Comment by celulamp** [comment]
> @celulamp all our servers are RP servers

I genuinely think that sybil or terry will either not care about this and gloss over it
or people will try to do funny stuff with it and then just get fucked over by other ppl doing many different sybil/terry hijinks

People also don't like to have to do these sort of things because they prefer making their own gimmicks or stuff up themselves.
Then again it's all optional but shrug i'm not sure if it's worth putting in all this effort and how much of a feature it will be used.

Sybil and Terry have different culture compared to manuel so certain prs won't work as well on them. (Idk why you always deflect this with saying that every server is a rp server because they are very different playstyles)



> > I'm also assuming these roles will have their own rules somewhere between a crewmembers and an antagonist, similar to lavaland where it's basically follow your flavor text.
> 
> That's up to the admins, we don't do policy here at the codebase, separation of server and codebase is absolute etc.
> 

I never understood this why should coders not think about how policy should work? That's what basically determines if your pr will get in or not (other than being good) If a PR is easy to implement into the policies we already have then it's more likely to get merged. But if it has to rework or add new policies then it probably won't get merged.

**Comment by Loafin34** [comment]
> terry and sybil won't like it
so, the only things terry and sybil will like are buffs to things they like and nerfs to things they don't like? It's fine to have a PR that is optional, and adds plot hooks to the station's story. I'm looking forward to it as an avid sybil/manuel player.

**Comment by Mothblocks** [comment]
> People also don't like to have to do these sort of things because they prefer making their own gimmicks or stuff up themselves.

*You* don't like it, which is why you can turn it off (and why I'm asking for it to be off by default). I don't see what the problem is with having additional features for people who want to roleplay with some direction?

**Comment by Mothblocks** [comment]
> I never understood this why should coders not think about how policy should work?

What Goof is replying to is this:

> I'm also assuming these roles will have their own rules somewhere between a crewmembers and an antagonist, similar to lavaland where it's basically follow your flavor text.

...which is not something Goof can decide on even if he wanted to because he is not a head admin. He can have suggestions but it would be disingenuous to act on the PR based on that. That being said whenever hand-off happens we can ping the new head admins to let them know this PR is coming.

**Comment by SomeoneYouProbablyKnow** [comment]
> I genuinely think that sybil or terry will either not care about this and gloss over it or people will try to do funny stuff with it and then just get fucked over by other ppl doing many different sybil/terry hijink.

This is the definition of a roleplay prompt. Someone participating in a story will impact you literally AS MUCH as someone doing their gimmick. If you don't want promts then set your prefs off.

If people doing stories/gimmicks get swept up in terry madness then good. The stories are not meant to play out like they have a tight script they are supposed to interact with the round at large.

**Comment by dox-net** [comment]
As I player, not that my opinion really matters that much but
this strays a little too far from what tg station is supposed to be, I play tg station for the simplicity, If i wanted hardcore rp with storylines and stuff, I would literally go somewhere else
IDK i just dont like this idea

**Comment by Iamgoofball** [comment]
@dox-net too bad

**Comment by SomeoneYouProbablyKnow** [comment]
> I play tg station for the simplicity.

I kinda get what you are saying but if you take a look at the stories themselves they ARE simple. They are essentially well written improv prompts for a shift gimmick.

**Comment by Mothblocks** [comment]
They're optional and off by default and for people who want to roleplay but have trouble coming up with stuff themselves, which is perfectly reasonable and a group I want to support

**Comment by optimumtact** [comment]
>  I would literally go somewhere else

I hear fulp station is pretty simple and fun, see you there

**Comment by jlsnow301** [comment]
> If i wanted hardcore rp with storylines and stuff, I would literally go somewhere else

These are purely optional, so I think a better way of looking at this is as extra assistance for players wanting to do gimmick roles

Otherwise they're just an assistant with a shop

**Comment by Wallemations** [comment]
Does this use a similar opt-in system as antags do, or does it pick from every crew member regardless?

**Comment by ArthuroPep** [comment]
> Does this use a similar opt-in system as antags do, or does it pick from every crew member regardless?

You can opt in like antags, turned off by default

**Comment by Loafin34** [comment]
Okay, read a little bit through, I think it'd be more thematically appropriate for the actors to have Space Cash, as they are rarely used, and should be from a bank, rather than from their ID. Also, my only problem with the Actors are the medical student and Businessman; it's referenced that they're young, so should their age be changed to be young?

**Comment by Iamgoofball** [comment]
ive seen some pretty fuckin old medical students in my time, but I'll see what I can do

**Comment by Loafin34** [comment]
Mafiosos would be fun if one had a baseball bat and the other a tommygun. Would be accurate.

**Comment by DrDiasyl** [comment]
I was never disappointed in you Goof.

**Comment by jlsnow301** [comment]
Hey there, this looks like it's stalled a bit. Perhaps it's better to divide this into multiple PRs. You can at least get some feedback and work from there.

**Comment by Addust** [comment]
i approve but I SWEAR TO GOD if i get a safety inspector questioning why my whiteship is firing out accelerator lasers while being powered by the shittiest SM setup ever designed i will blow their face off

**Comment by Loafin34** [comment]
goof.... it's been the weekend where is the work, please do SOMETHING with it I do not want to see stories and actors die

**Comment by Iamgoofball** [comment]
i promise i'll work on it ok

**Comment by Iamgoofball** [comment]
guess who's still busy with work lmao

**Comment by Mothblocks** [comment]
please ask a maintainer to reopen when this is ready, we have too many prs on the tracker

**Comment by Loafin34** [comment]
goof it's been a month

**Comment by Quink-Fix** [comment]
goof please

**Comment by Iamgoofball** [comment]
i have a life and work

**Comment by Loafin34** [comment]
@Iamgoofball please god work on this over the weekend

**Pull request closed without merging**
